### PR TITLE
New settings: implement verify secondary email

### PR DESCRIPTION
**Pull request by dannycoates**
This adds the ui to add a secondary email and verify it with a code.

It adds the `/emails` and `emails/verify` routes. I used `/emails` in order to match the existing settings route.

### CODE REVIEW

File: packages/fxa-settings/src/components/FlowSecondaryEmailAdd/index.tsx [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+          account: (existing: Account) => {
+            const emails = [
+              ...existing.emails,
+              { email: email!, isPrimary: false, verified: false },
+            ];
+            return { ...existing, emails };
+          },
+        },
+      });
+    },
```
Comment (dannycoates) [inline packages/fxa-settings/src/components/FlowSecondaryEmailAdd/index.tsx:47]: @jodyheavener look here

File: packages/fxa-settings/src/components/FlowSecondaryEmailVerify/index.tsx [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+      cache.modify({
+        fields: {
+          account: (existing: Account) => {
+            const emails = existing.emails.map((m) =>
+              m.email === email ? { ...m, verified: true } : { ...m }
+            );
+            return { ...existing, emails };
+          },
+        },
+      });
```
Comment (dannycoates) [inline packages/fxa-settings/src/components/FlowSecondaryEmailVerify/index.tsx:44]: @jodyheavener this one

File: packages/fxa-settings/src/components/FlowSecondaryEmailVerify/index.tsx [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+`;
+
+export const FlowSecondaryEmailVerify = ({ location }: RouteComponentProps) => {
+  const [code, setCode] = useState<string>();
+  const [errorText, setErrorText] = useState<string>();
+  const goBack = useCallback(() => window.history.back(), []);
+  const navigate = useNavigate();
+  const email = (location?.state as any)?.email as string | undefined;
+
+  const [verifySecondaryEmail] = useMutation(VERIFY_SECONDARY_EMAIL_MUTATION, {
```
Comment (dannycoates) [inline packages/fxa-settings/src/components/FlowSecondaryEmailVerify/index.tsx:25]: I had trouble using the new `useHandledMutation` function. The `setErrorText` wasn't triggering the ui to update. I haven't debugged it yet.

Comment (jodyheavener) [reply packages/fxa-settings/src/components/FlowSecondaryEmailVerify/index.tsx:25]: Interesting - FWIW I have it working in a similar fashion here https://github.com/mozilla/fxa/pull/6292/files#diff-a313ff76f12b71cf55da0fc953b5dabbR88-R91

Comment (jodyheavener) [reply packages/fxa-settings/src/components/FlowSecondaryEmailVerify/index.tsx:25]: Fixed it, it was calling itself https://github.com/mozilla/fxa/pull/6292/files#diff-ed45386ca794096a72b565395c90232dR75-R89

**Pull request merged**